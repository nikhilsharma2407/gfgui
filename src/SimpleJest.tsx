import { useState } from "react";

export default function SimpleJest() {
    const [name,setName] = useState("");
    return(
        <>
            <input type="text" data-testid="input" onChange={(e) => setName(e.target.value)} />
            <label data-testid="output">{`hello ${name}`}</label> 
        </>
    );
}