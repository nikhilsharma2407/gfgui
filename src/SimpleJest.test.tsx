import { fireEvent, render } from "@testing-library/react"
import SimpleJest from "./SimpleJest"

describe("SimpleJest", () => {
    it("Simple test",() => {
        const {baseElement} =render(<SimpleJest />)
        expect(baseElement).toBeInTheDocument();
    })

    it("text changes", () => {
        const obj = {target:{value: "geeks"}};
        const {getByTestId} = render(<SimpleJest />);
        const input = getByTestId("input");
        fireEvent.change(input,obj);
        const output = getByTestId("output");
        expect(output.innerHTML).toBe("hello geeks");
    })

    it("snapshot testing", () => {
        const {baseElement} = render(<SimpleJest />);
        expect(baseElement).toMatchSnapshot();
    })
})