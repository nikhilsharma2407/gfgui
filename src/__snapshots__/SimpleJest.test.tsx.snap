// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleJest snapshot testing 1`] = `
<body>
  <div>
    <input
      data-testid="input"
      type="text"
    />
    <label
      data-testid="output"
    >
      hello 
    </label>
  </div>
</body>
`;
